File: leetcode/js/numWaterBottles.js
/**
 * @param {number} numBottles
 * @param {number} numExchange
 * @return {number}
 */
const numWaterBottles = function(numBottles, numExchange) {
  let result = 0;

  while (numBottles > 0) {
    if (numBottles < numExchange)  {
      result += numBottles
      break
    }
    result += numExchange;
    numBottles -= numExchange - 1
  }
  return result
};

console.log(numWaterBottles(9, 3))
console.log(numWaterBottles(15, 4))
console.log(numWaterBottles(2,3))

File: leetcode/java/src/com/company/math/numWaterBottles/numWaterBottles.java
package com.company.math.numWaterBottles;

class Solution {
  public int numWaterBottles(int numBottles, int numExchange) {
    int res = 0;

    while (numBottles > 0) {
      if (numBottles < numExchange) {
        res += numBottles;
        break;
      }
      res += numExchange;
      numBottles -= numExchange - 1;
    }

    return res;
  }
}